pkg/apiserver/apiserver.go

```diff
@@ -194,45 +194,68 @@ func getHistory(request *restful.Request, response *restful.Response) {
 		s := fmt.Sprintf("Could not find any provenance history for resource name: %s", resourceName)
 		response.Write([]byte(s))
 	} else {
-		//TODO: handle optional interval parameters
-		response.Write([]byte(intendedProvObj.ObjectFullHistory.SpecHistory()))
+		if start != "" && end != "" { //have both a start and an end
+			fmt.Printf("Start:%s", start)
+			fmt.Printf("End:%s", end)
+			startInt, err := strconv.Atoi(start)
+			if err != nil {
+				s := fmt.Sprintf("Could not parse start query parameter to int: %s", err.Error())
+				response.Write([]byte(s))
+				return
+			}
+			endInt, err := strconv.Atoi(end)
+			if err != nil {
+				s := fmt.Sprintf("Could not parse end query parameter to int: %s", err.Error())
+				response.Write([]byte(s))
+				return
+			}
+			fmt.Printf("Spec history starting with version %d and ending with version %d", startInt, endInt)
+			response.Write([]byte(intendedProvObj.ObjectFullHistory.SpecHistoryInterval(startInt, endInt)))
+		} else { //start and end
+			response.Write([]byte(intendedProvObj.ObjectFullHistory.SpecHistory()))
+		}
 	}
 
 }
 
 func bisect(request *restful.Request, response *restful.Response) {
 	fmt.Println("Inside bisect")
 	resourceName := request.PathParameter("resource-id")
-	requestPath := request.Request.URL.Path
-	resourcePathSlice := strings.Split(requestPath, "/")
-	resourceKind := resourcePathSlice[6] // Kind is 7th element in the slice
-
-	var provenanceInfo string
-	provenanceInfo = "Resource Name:" + resourceName + " Resource Kind:" + resourceKind
-	fmt.Println(provenanceInfo)
-
-	field := request.QueryParameter("field")
-	value := request.QueryParameter("value")
-
-	provenanceInfo = provenanceInfo + " Field:" + field + "Value: " + value + "\n"
-
-	fmt.Printf("ProvenanceInfo:%v", provenanceInfo)
-
-	response.Write([]byte(provenanceInfo))
-
+	requestPath := request.Request.URL.String()
+	// resourcePathSlice := strings.Split(requestPath, "/")
+	// resourceKind := resourcePathSlice[6] // Kind is 7th element in the slice
+	// fmt.Println(requestPath)
+	strs := strings.Split(requestPath, "/")
+	// assuming that the last slash is where the query starts.
+	// so the URL parameters are rearranged alphabetically
+	// apis/kubeprovenance.cloudark.io/v1/namespaces/default/postgreses/client25/bisect?field1=username&field2=password&value1=pallavi&value2=pass123
+	// field1=username&field2=password&value1=pallavi&value2=pass123
+	args := strings.Split(strs[len(strs)-1], "?")[1] //get rid of bisect?
+	argMap := make(map[string]string)
+	argsArray := strings.Split(args, "&")
+	for _, val := range argsArray {
+		fieldToValue := strings.Split(val, "=")
+		argMap[fieldToValue[0]] = fieldToValue[1]
+	}
+	// var provenanceInfo string
+	// provenanceInfo = "Resource Name:" + resourceName + " Resource Kind:" + resourceKind
+	// fmt.Println(provenanceInfo)
 	intendedProvObj := provenance.FindProvenanceObjectByName(resourceName, provenance.AllProvenanceObjects)
 	if intendedProvObj == nil {
```

Comment:
I am not following this if else block (lines 254 - 261).

Why do we try to first FindProvenance for ResourceName and if there is nothing call Bisect?